src/emulate.c

```diff
@@ -395,36 +397,47 @@ enum {
 #undef _
 };
 
+/* FIXME: This will simply find the n-th instruction by iterating
+ * the linked list linearly, we may want to find better approach. */
+#define GET_NEXT_N_INSN(ir, n)                     \
+    ({                                             \
+        rv_insn_t *__tmp = ir;                     \
+        for (typeof(n) iter = 0; iter < n; iter++) \
```

Comment:
Check `src/common.h` for the awareness of `typeof`. That is, you shall provide non-typeof path for this proposed macro.